ORM/DataList.php

```diff
@@ -550,30 +537,40 @@ protected function isValidRelationName($field) {
 	}
 
 	/**
-	 * Translates a filter type to a SQL query.
+	 * Given an input string, parse it into the field name, filter type, and modifiers as
+	 * a trio.
 	 *
-	 * @param string $field - the fieldname in the db
-	 * @param string $filter - example StartsWith, relates to a filtercontext
-	 * @param array $modifiers - Modifiers to pass to the filter, ie not,nocase
-	 * @param string $value - the value that the filtercontext will use for matching
-	 * @return static
+	 * @param string $filter E.g. `Name:ExactMatch:not`, `Name:ExactMatch`, `Name:not`, `Name`
+	 * @return array List with the items field name, filter service, and list any of modifiers
 	 */
-	private function applyFilterContext($field, $filter, $modifiers, $value) {
-		if($filter) {
-			$className = "{$filter}Filter";
-		} else {
-			$className = 'ExactMatchFilter';
-		}
+	protected function parseFilterArguments($filter) {
+		// Field name is always the first component
+		$fieldArgs = explode(':', $filter);
+		$fieldName = array_shift($fieldArgs);
 
-		if(!class_exists($className)) {
-			$className = 'ExactMatchFilter';
-
-			array_unshift($modifiers, $filter);
+		// Inspect type of second argument to determine context
+		$secondArg = array_shift($fieldArgs);
+		$modifiers = $fieldArgs;
+		if(!$secondArg) {
+			// Use default filter if none specified. E.g. `->filter(['Name' => $myname])`
+			$filterServiceName = 'DataListFilter.default';
+		} else {
+			// The presence of a second argument is by default ambiguous; We need to query
+			// Whether this is a valid modifier on the default filter, or a filter itself.
+			/** @var SearchFilter $defaultFilterInstance */
+			$defaultFilterInstance = Injector::inst()->get('DataListFilter.default');
+			if (in_array(strtolower($secondArg), $defaultFilterInstance->getSupportedModifiers())) {
+				// Treat second (and any subsequent) argument as modifiers, using default filter
+				$filterServiceName = 'DataListFilter.default';
+				array_unshift($modifiers, $secondArg);
+			} else {
+				// Second argument isn't a valid modifier, so assume is filter identifier
+				$filterServiceName = "DataListFilter.{$secondArg}";
```

Comment:
Do you think we should use `Injector::inst()->hasService($filterServiceName)` to throw an error if a service isn't defined?